templates/submission/row.html

```diff
@@ -49,9 +49,13 @@
             <i class="fa fa-eye fa-fw"></i><span class="label">{{ _('view') }}</span>
         </a>
         {% if perms.judge.rejudge_submission and can_edit %} ·
-            <a href="#" onclick="rejudge_submission({{ submission.id }}, event);return false">
-                <i class="fa fa-refresh fa-fw"></i><span class="label">{{ _('rejudge') }}</span>
-            </a>
+            {% if not submission.is_locked %}
+                <a href="#" onclick="rejudge_submission({{ submission.id }}, event);return false">
+                    <i class="fa fa-refresh fa-fw"></i><span class="label">{{ _('rejudge') }}</span>
+                </a>
+            {% else %}
+                <i class="fa fa-refresh fa-fw"></i><span class="grey-label">{{ _('locked') }}</span>
```

Comment:
I think we should also add some alt text here, like "This submission has been locked, and cannot be rejudged."